Automattic/wp-calypso
# Stats: Updating UTM module with empty states

Related to https://github.com/Automattic/red-team/issues/72

## Proposed Changes

* adding new empty state for UTM
* extending UTM Builder to accept a custom trigger
* unity TS types for the new modules

![SCR-20240702-mpbt](https://github.com/Automattic/wp-calypso/assets/112354940/d68a5e6f-4d8b-4798-bfc4-95268151d766)

## Why are these changes being made?


* a part of the empty states project

## Testing Instructions



* go to the live branch
* apply `stats/empty-module-traffic` feature flag
* verify that a page with an empty UTM component works with and without the feature flag
* repeat for blogs with data
* verify that the advanced feature overlay works properly
* verify that the new empty state only shows up after applying the feature flag

## Pre-merge Checklist
